gh-89083: add support for UUID version 7 (RFC 9562) (#121119)
python/cpython: Doc/library/uuid.rst

--------------

> This module provides immutable :class:`UUID` objects (the :class:`UUID` class) and
> the :ref:`functions <uuid-factory-functions>` for generating UUIDs corresponding

Reviewer: ```suggestion
This module provides immutable :class:`UUID` objects (the :class:`UUID` class)
and :ref:`functions <uuid-factory-functions>` for generating UUIDs corresponding
```

> This module provides immutable :class:`UUID` objects (the :class:`UUID` class) and
> the :ref:`functions <uuid-constructor-functions>` for generating UUIDs corresponding

Reviewer: ```suggestion
This module provides immutable :class:`UUID` objects (the :class:`UUID` class)
and :ref:`functions <uuid-functions>` for generating UUIDs corresponding
```

«constructor» has a different meaning related to class, these are rather factory functions: they give a simplified interface to instantiate a class. But I suggest that just saying functions is good enough.

Author: I will name them `uuid-factory-functions`. (The `uuid` prefix is for the module, not for the fact that we're creating UUID objects; it's to disambiguate Sphinx labels)

to a specific UUID version as specified in :rfc:`9562` (which supersedes :rfc:`4122`),
e.g., :func:`uuid1` for UUID version 1, :func:`uuid3` for UUID version 3, and so on.

> Note that UUID version 2 is deliberately omitted as it exceeds the scope of the RFC.

Author: ```suggestion
Note that UUID version 2 is deliberately omitted as it is outside the scope of the RFC.
```

> Note that UUID version 2 is deliberately omitted as it is outside the scope of the RFC.

Author: ```suggestion
Note that UUID version 2 is deliberately omitted as it exceeds the scope of the RFC.
```

Reviewer: This change is incorrect IMO.

Author: Right, I should have trusted my first wording...

--------------

This module provides immutable :class:`UUID` objects (the :class:`UUID` class)
and :ref:`functions <uuid-factory-functions>` for generating UUIDs corresponding
to a specific UUID version as specified in :rfc:`9562` (which supersedes :rfc:`4122`),

> e.g., :func:`uuid1` for UUID version 1, :func:`uuid3` for UUID version 3, and so on.

Reviewer 2: https://devguide.python.org/documentation/style-guide/#use-simple-language

```suggestion
for example, :func:`uuid1` for UUID version 1, :func:`uuid3` for UUID version 3, and so on.
```

Author: I'm sorry, I'm too used to that formulation :')

This module provides immutable :class:`UUID` objects (the :class:`UUID` class)
and the functions :func:`uuid1`, :func:`uuid3`, :func:`uuid4`, :func:`uuid5`,

> :func:`uuid6`, :func:`uuid7`, and :func:`uuid8` for generating UUIDs version 1,
> 3, 4, 5, 6, 7, and 8 as specified in :rfc:`9562` (which supersedes :rfc:`4122`).

Reviewer 2: ```suggestion
:func:`uuid6`, :func:`uuid7`, and :func:`uuid8` for generating UUIDs for version 1
and 3-8 as specified in :rfc:`9562` (which supersedes :rfc:`4122`).
```

or we don't need to repeat the numbers again:

```suggestion
:func:`uuid6`, :func:`uuid7`, and :func:`uuid8` for generating UUIDs for those versions,
as specified in :rfc:`9562` (which supersedes :rfc:`4122`).
```

Author: Oh. I like the second formulation! WDYT @merwok?

Reviewer: We are in the first lines of the doc. For readers it could be the first introduction to the concept of UUID versions, so I prefer the inelegant but clear enumeration (previous text) than a compact and less clear phrasing («for those versions» or «uuidN»).

But that doesn’t mean we need to write 1, 3, 4, 5, 6, 7, 8. Something better could like:
This modules provides a UUID class and functions for generating UUIDs corresponding to a specific UUID version as specified in the RFC: uuid1 for version 1, uuid3, uuid4, uuid5, [etc].

(I think enumerating the functions was done to provide a quick link to jump there, but we could replace that with a single link going to `The uuid module defines the following functions`.)

Author: > For readers it could be the first introduction to the concept of UUID versions

I haven't thought about this so thanks for the feedback. I think we can go with:

```rst
This modules provides a :class:`UUID` class and :ref:`functions <uuid-functions>` for
generating UUIDs corresponding to a specific UUID version as specified in the RFC, e.g.,
:func:`uuid1` for UUID version 1, :func:`uuid3` for UUID version 3, and so on (note that
UUID version 2 is deliberately omitted as it is outside the scope of the RFC).
```

      globally unique, while the latter are not.

> .. _uuid-constructor-functions:

Reviewer: ```suggestion
.. _uuid-functions:
```
